.github/workflows/java-17-builds.yml

```diff
@@ -1,26 +1,29 @@
-name: Java CI
+name: Java 17 CI (MC 1.17+)
 
 on: [push, pull_request]
 
 jobs:
     build:
         if: "! contains(toJSON(github.event.commits.*.message), '[ci skip]')"
         runs-on: ubuntu-latest
-
         steps:
-            - uses: actions/checkout@v2
+            - uses: actions/checkout@v3
               with:
                   submodules: recursive
             - name: Set up JDK 17
-              uses: actions/setup-java@v1
+              uses: actions/setup-java@v2
               with:
-                  java-version: 17
+                  java-version: '17'
+                  distribution: 'adopt'
+                  cache: gradle
+            - name: Grant execute permission for gradlew
+              run: chmod +x gradlew
```

Comment:
Why was this needed now, it had always worked without it

Reply:
I've had issues in the past where actions would error because of it. Maybe GitHub automatically applies it now? I'm not taking any chances.